### PR TITLE
Implement function to look for project root dir

### PR DESCRIPTION
**Pull request by jcarpanelli**
Fixes https://github.com/zeppelinos/zos/issues/646. Also, it could be useful for https://github.com/zeppelinos/zos/pull/792 and running `zos init` anywhere in the project :eyes:.

### CODE REVIEW

File: packages/cli/src/bin/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import fs from 'fs';
+
+export default function lookForRootDirectory(dirPath: string): string | null {
+  if (dirPath.length === 0) return null;
+  const dirFiles = fs.readdirSync(dirPath);
+
+  if ((dirFiles.includes('package.json') && dirFiles.includes('package-lock.json')) || dirFiles.includes('zos.json')) {
+    return dirPath;
+  } else {
+    const dir = dirPath.split('/');
```
Comment (ylv-io) [inline packages/cli/src/bin/helpers.ts]: Not every OS use '/' as path separator. User [path.sep](https://nodejs.org/api/path.html#path_path_sep) instead.

Comment (jcarpanelli) [reply packages/cli/src/bin/helpers.ts]: Good catch!

**Comment by jcarpanelli** [comment]
> On a side note. What is the reason for us writing these helpers when there is a [library](https://www.npmjs.com/package/find-up) with the same functionality and 16kk downloads a week?
> Like `(await findUp(['package.json', 'package-lock.json', 'zos.json']))`?

I didn't know about the existence of that library! Thanks :heart: 
I double-checked it, and it seems that `find-up#findUp` function will return the full path of the first occurrence (e.g., `/home/foo/project/package.json`), so we still have to remove the file name from the string returned. 
I also noticed that it depends on 5 other packages (directly or indirectly, see [the dependency tree](http://npm.broofa.com/?q=find-up)), which could be overkill compared to the function implemented in this PR. Anyways, I will probably replace it later today.

File: packages/cli/src/bin/zos-cli.ts [hunk 2]
```diff
@@ -3,11 +3,15 @@
...
+import lookForRootDirectory from './helpers';
+
+const rootDirectory = lookForRootDirectory(process.cwd()) || process.cwd();
+if (rootDirectory !== null && rootDirectory !== process.cwd()) process.chdir(rootDirectory);
 
 // Acquire file lock to ensure no other instance is running
 const LOCKFILE: string = '.zos.lock';
 try {
-  lockSync(LOCKFILE, { retries: 0 });
+  lockSync(`${rootDirectory}/${LOCKFILE}`, { retries: 0 });
```
Comment (spalladino) [inline packages/cli/src/bin/zos-cli.ts]:
```suggestion
  lockSync(LOCKFILE, { retries: 0 });
```
Should not be needed, as we are doing a chdir right before

**Comment by spalladino** [comment]
As discussed offline, let's add an exception when the user is explicitly running `zos init` to ensure is run on the current folder. Otherwise, we make it impossible to create a new project from scratch using `zos init` in a current folder if there is a `package.json` up in the tree (as is, for instance, in a monorepo).

**Comment by jcarpanelli** [comment]
> As discussed offline, let's add an exception when the user is explicitly running `zos init` to ensure is run on the current folder. Otherwise, we make it impossible to create a new project from scratch using `zos init` in a current folder if there is a `package.json` up in the tree (as is, for instance, in a monorepo).

I've just pushed this change together with @ylv-io suggestions. Ready for review!

File: packages/cli/src/bin/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import path from 'path';
+import findUp from 'find-up';
+
+export default function findRootDirectory(cwd: string): string | null {
+  const filePath = findUp.sync(['package.json', 'zos.json'], { cwd });
+  if (!filePath) return null;
+  const dir = filePath.split(path.sep);
```
Comment (ylv-io) [inline packages/cli/src/bin/helpers.ts]: My two cents.
There is a method [path.dirname](https://nodejs.org/api/path.html#path_path_dirname_path) so you can do `return path.dirname(filePath);`. Feels more natural.

**Comment by jcarpanelli** [comment]
I've just implemented @ylv-io last suggestion :smile: 

@spalladino I also removed all calls to the `init` command. It will be very easy to add them again later. I think we should further discuss how we can include that feature, as I feel that just running `zos create` anywhere in the project and doing everything under the hood (_init_, add, push) will be the easiest, cleanest and more awesome way of creating a proxy without bothering the user with unnecessary steps.

That being said, we can think how to readd this later and merge this PR now.

**Comment by spalladino** [comment]
Agree, I think we're good to merge. @ylv-io could you update your review?

**Pull request merged**